kore/src/Kore/Step/EquationalStep.hs

```diff
@@ -432,28 +436,59 @@ matchRule
     -- ^ Rule
     -> unifier (UnifiedRule variable (rule variable))
 matchRule sideCondition initial rule = do
-    let (initialTerm, initialCondition) = Pattern.splitTerm initial
-    -- Unify the left-hand side of the rule with the term of the initial
-    -- configuration.
-    let ruleLeft = matchingPattern rule
-    unification <- unifyPatterns ruleLeft initialTerm >>= maybe empty return
+    unification <- unifyPatterns >>= maybe empty return
     -- Combine the unification solution with the rule's requirement clause,
     let requires = precondition rule
     unification' <-
         evaluateRequires sideCondition initialCondition unification requires
+    checkConcrete concreteVars symbolicVars unification'
```

Comment:
We should not evaluate the `requires` if the required variables are not symbolic or concrete, so this check should move up.

Reply:
Well, it was up; however, we need to normalize the substitution before the check and this means doing a predicate simplification.  We could probably perform two simplifications, one before, with just substitution, then perform the checks, then add the requires and resimplify.

Would you rather have that?

Reply:
The matcher _always_ returns a normalized substitution, it should not be necessary to do any simplification after.